.github/workflows/compilation_on_macos.yml

```diff
@@ -214,33 +217,68 @@ jobs:
         working-directory: product-mini/platforms/${{ matrix.platform }}
 
   build_samples_wasm_c_api:
-    needs: [build_iwasm]
+    needs:
+      [
+        build_iwasm,
+        build_llvm_libraries_on_intel_macos,
+        build_wamrc,
+      ]
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         make_options: [
-            # Running modes supported
+            $AOT_BUILD_OPTIONS,
             $CLASSIC_INTERP_BUILD_OPTIONS,
             $FAST_INTERP_BUILD_OPTIONS,
-            # Running modes unsupported
-            #$LLVM_LAZY_JIT_BUILD_OPTIONS,
-            #$LLVM_EAGER_JIT_BUILD_OPTIONS,
-            #$AOT_BUILD_OPTIONS,
+            $FAST_JIT_BUILD_OPTIONS,
+            $LLVM_LAZY_JIT_BUILD_OPTIONS,
+            $LLVM_EAGER_JIT_BUILD_OPTIONS,
+            $MULTI_TIER_JIT_BUILD_OPTIONS,
           ]
         os: [macos-13]
+        include:
+          - os: macos-13
```

Comment:
Since macOS is not my primary development environment, and I learned that macOS's latest products and OS (after version 13) are all based on ARM chips, when comparing macOS-13 and macOS-1x (after 13), which OS is most popular, and which OS do you think WAMR CI and binary releases should target?

Reply:
i don't know which of arm/intel are popular at this point.
i guess it's better to provide both. (or universal binary)
i myself have no arm-based mac.

Reply:
> or universal binary

to follow https://developer.apple.com/documentation/apple-silicon/building-a-universal-macos-binary ?